src/test/java/com/mastery/java/task/rest/EmployeeControllerTest.java

```diff
@@ -1,4 +1,189 @@
+
+        this.mockMvc.perform(post("/"+id+"/edit")
+                .param("firstName", employee.getFirstName())
+                .param("lastName", employee.getLastName())
+                .param("departamentId", String.valueOf(employee.getDepartamentId()))
+                .param("jobTitle", employee.getJobTitle())
+                .param("gender", String.valueOf(employee.getGender()))
+                .param("dateOfBirth", String.valueOf(employee.getDateOfBirth()))).andDo(print())
+                .andExpect(MockMvcResultMatchers.redirectedUrl("/"));
+        then(service).should().update(employee);
+    }
+
+    @Test
+    public void deleteEmployeeTest() throws Exception {
+        Employee employee = stubEmployee();
+        int id = employee.getEmployeeId();
+        this.mockMvc.perform(post("/"+id+"/delete")).andDo(print())
+                .andExpect(MockMvcResultMatchers.redirectedUrl("/"));
+        then(service).should().deleteById(id);
+    }
+
+    private List<Employee> stubEmployeeList() {
+        List<Employee> employeeList = new ArrayList<Employee>();
+
+        int[] employeeId = new int[]{1, 2, 3};
+        String[] firstName = new String[]{"Andy", "Joe Lo", "Chelsea"};
+        String[] lastName = new String[]{"Samberg", "Truglio", "Peretti"};
+        int[] departamentId = new int[]{1, 2, 3};
+        String[] jobTitle = new String[]{"Java Developer", "HR", "Tester"};
+        Gender[] gender = new Gender[]{MALE, MALE, FEMALE};
+        Date[] date = new Date[3];
+        String[] datestr = new String[]{"1989-03-02", "1993-06-05", "1995-08-07"};
+        for (int i = 0; i < 3; i++)
+        date[i] = java.sql.Date.valueOf(datestr[i]);
+
+        for (int i = 0; i < 3; i++) {
+            Employee emp = new Employee();
+            emp.setEmployeeId(employeeId[i]);
+            emp.setFirstName(firstName[i]);
+            emp.setLastName(lastName[i]);
+            emp.setDepartamentId(departamentId[i]);
+            emp.setJobTitle(jobTitle[i]);
+            emp.setGender(gender[i]);
+            emp.setDateOfBirth(date[i]);
+            employeeList.add(emp);
+        }
+        return employeeList;
+    }
+
+    private Employee stubEmployee(){
+        Employee employee = new Employee();
+        employee.setEmployeeId(employeeId);
+        employee.setFirstName(firstName);
+        employee.setLastName(lastName);
+        employee.setDepartamentId(departamentId);
+        employee.setJobTitle(jobTitle);
+        employee.setGender(gender);
+        employee.setDateOfBirth(date);
+        return employee;
+    }
+
+    private Employee stubEmployeeWithoutId(){
+        Employee employee = new Employee();
+        employee.setFirstName(firstName);
+        employee.setLastName(lastName);
+        employee.setDepartamentId(departamentId);
+        employee.setJobTitle(jobTitle);
+        employee.setGender(gender);
+        employee.setDateOfBirth(date);
+        return employee;
+    }
```

Comment:
maybe it would be better to use builder?

Reply:
Of corse, done